pilot/pkg/model/context.go

```diff
@@ -154,6 +230,14 @@ func (e *Environment) SetLedger(l ledger.Ledger) {
 	e.ledger = l
 }
 
+// only user for testing
+func (e *Environment) SetClusterLocalHosts(hostList []string) {
+	e.clusterLocalHosts = make([]host.Name, len(hostList))
```

Comment:
And here ... maybe you should create the list first and then just lock when you assign it to `e.clusterLocalHosts`.